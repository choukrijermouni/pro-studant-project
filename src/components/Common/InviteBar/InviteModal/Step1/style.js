import {
  neutral,
  blue,
  red
} from '@pro_boa/ui'

export default {
  modal: {
    width: 580,
    backgroundColor: neutral[0],
    borderRadius: 4,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  header: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    alignItems: 'center'
  },
  avatars: {
    display: 'flex'
  },
  userImage: {
    marginLeft: -18
  },
  input: {
    width: 200
  },
  footer: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  invitesBox: {
    height: 72,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center ',
    alignItems: 'center',
    borderRadius: 4,
    backgroundColor: neutral[0],
    boxShadow: '0 0 36px 0 rgba(194,212,230,0.23)',
    transition: 'transform 0.2s ease-in-out',
    '&:hover': {
      transform: 'scale(1.03)'
    }
  },
  invites: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center ',
    alignItems: 'center'
  },
  icon: {
    fontSize: 24,
    color: blue[1]
  },
  editIcon: {
    fontSize: 15,
    color: neutral[3],
    padding: '13px 12px',
    border: `1px solid ${neutral[3]}`,
    borderRadius: '50%',
    marginLeft: 14,
    cursor: 'pointer',
    '&:hover': {
      color: blue[1],
      border: `1px solid ${blue[1]}`
    }
  },
  licenseButtons: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  line: {
    backgroundColor: neutral[13],
    height: 1,
    border: 'none',
    marginBottom: 20
  },
  circle: {
    borderRadius: '50%',
    width: 17,
    height: 17,
    padding: 2,
    background: red[2],
    color: neutral[0],
    textAlign: 'center',
    font: '12px Arial, sans-serif',
    left: -10,
    top: -3,
    position: 'relative'
  },
  noLicenseMessageContainer: {
    width: '70%',
    marginRight: 25
  },
  image: {
    maxWidth: 156
  }
}
