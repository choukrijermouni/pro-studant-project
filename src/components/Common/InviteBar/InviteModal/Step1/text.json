{
  "inviteTitle": "Inviter des apprenants",
  "subHeader": {
    "first": "Vous avez invité ",
    "second": "nouveaux apprenants à rejoindre votre organisme"
  },
  "learner": "Apprenants",
  "licenseAffect": "Affectez des licences à vos apprenants",
  "monthlyLicenseLeft": "Licences mensuelles disponibles : ",
  "yearlyLicenseLeft": "Licences annuelles disponibles : ",
  "trialLicenseLeft": "Licences d'essai disponibles : ",
  "allTeams": "Équipes",
  "withoutButton": "Aucune",
  "trialButton": "Essai",
  "yearlyButton": "Annuelle",
  "monthlyButton": "Mensuelle",
  "confirmButton": "Confirmer",
  "abortButton": "Annuler",
  "withoutTeam": "Sans",
  "noLicenseLeftMessage": "Vous avez épuisé toutes les licences. Veuillez contacter notre équipe pour vous prendre en charge",
  "noLicenseMessage": "Aucune licence disponible. Veuillez contacter notre équipe",
  "on": "par e-mail à l’adresse :",
  "or": "ou par téléphone au :",
  "forGettingNewLicenses": "pour commander de nouvelles licences."
}