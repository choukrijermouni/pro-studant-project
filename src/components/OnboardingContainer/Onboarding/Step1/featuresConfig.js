import {
  fastActions,
  analytics,
  createTeams,
  dashboard,
  reports,
  transferLicense
} from 'assets'
import {
  red,
  yellow,
  purple,
  blue,
  magenta,
  green
} from 'constants/'
import Text from './text.json'

export default [
  {
    id: 1,
    text: Text.features.feature1,
    icon: dashboard,
    color: yellow
  },
  {
    id: 2,
    text: Text.features.feature2,
    icon: analytics,
    color: red
  },
  {
    id: 3,
    text: Text.features.feature3,
    icon: fastActions,
    color: purple
  },
  {
    id: 4,
    text: Text.features.feature4,
    icon: reports,
    color: blue
  },
  {
    id: 5,
    text: Text.features.feature5,
    icon: transferLicense,
    color: magenta
  },
  {
    id: 6,
    text: Text.features.feature6,
    icon: createTeams,
    color: green
  }
]
