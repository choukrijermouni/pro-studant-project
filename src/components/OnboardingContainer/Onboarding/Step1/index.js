import {
  Paper,
  Button,
  TypographyElement
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import styles from './style'
import { logo as pro_boa } from 'assets/index'
import Text from './text.json'
import FeatureCard from 'components/OnboardingContainer/FeatureCard'
import features from './featuresConfig'
import { chunk } from 'lodash'
import { onboardingSteps } from 'constants/'

const useStyle = createUseStyles(styles)

const splitFeatures = chunk(features, 2).map(chunkFeatures => {
  chunkFeatures.push(...Array(2 - chunkFeatures.length))
  return chunkFeatures
})
export default ({ setStep }) => {
  const { paper, content, logo, line, licenseButtons, body, view, cardDeck } = useStyle()
  return (
    <div className={body}>
      <Paper className={paper}>
        <div className={content}>
          <img src={pro_boa} alt='notification' className={logo} />
          <TypographyElement
            component='h2'
            variant='heading2'
            align='left'
            spacing='40px 0 8px'
          >
            {Text.title}
          </TypographyElement>
          <TypographyElement
            component='h4'
            variant='body2'
            align='left'
            spacing='8px 0'
          >
            {Text.subHeader.first}{Text.subHeader.second}
          </TypographyElement>
          <TypographyElement
            component='h4'
            variant='heading4'
            align='left'
            spacing='8px 0'
          >
            {Text.subTitle}
          </TypographyElement>
          <div className={cardDeck}>
            {
              splitFeatures.map((card, key) =>
              (
                <div key={key} className={view}>
                  {
                    card.map((feature, id) => (
                      feature !== undefined
                        ? <FeatureCard key={id} cardIcon={feature.icon} context={feature.text} cardColor={feature.color} width={539} />
                        : <div />
                    ))
                  }
                </div>))
            }
          </div>
          <hr className={line} />
          <div className={licenseButtons}>
            <Button
              handleClick={() => setStep(onboardingSteps.newTraining)}
              width={153}
              label={Text.discoverButton}
            />
          </div>
        </div>
      </Paper>
    </div>
  )
}
