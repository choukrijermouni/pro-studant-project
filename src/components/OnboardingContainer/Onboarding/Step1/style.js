import { neutral } from '@pro_boa/ui'
import { headerBackground } from 'assets/index'

export default {
  content: {
    display: 'block',
    padding: '65px 48px'
  },
  logo: {
    height: 38
  },
  line: {
    backgroundColor: neutral[13],
    height: 1,
    border: 'none',
    marginBottom: 20
  },
  licenseButtons: {
    display: 'flex',
    justifyContent: 'end'
  },
  body: {
    backgroundImage: `url(${headerBackground})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'right top, right top',
    backgroundSize: 270,
    backgroundColor: neutral[0],
    borderRadius: 4,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  paper: {
    display: 'flex',
    flexDirection: 'column'
  },
  view: {
    display: 'flex',
    justifyContent: 'space-between',
    columnGap: 35
  },
  cardDeck: {
    display: 'flex',
    flexDirection: 'column',
    rowGap: 16,
    margin: '16px 0 34px'
  }
}
