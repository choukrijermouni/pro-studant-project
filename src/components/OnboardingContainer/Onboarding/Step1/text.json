{
  "title": " Nouvelles fonctionnalités disponibles sur votre nouvel espace Pro",
  "subHeader": {
    "first": "Bonjour",
    "second": ", pro_boa innove à nouveau et lance une nouvelle version de l’espace Pro pour améliorer votre quotidien."
  },
  "subTitle": " Fonctionnalités et améliorations",
  "discoverButton": "Découvrir",
  "abortButton": "Passer cette étape",
  "features": {
    "feature1": {
      "title": "Tableau de bord",
      "body": ""
    },
    "feature2": {
      "title": "Analytics",
      "body": ""
    },
    "feature3": {
      "title": "Actions rapides",
      "body": "Invitez vos apprenants et affectez des licences en toute simplicité."
    },
    "feature4": {
      "title": "Rapports",
      "body": ""
    },
    "feature5": {
      "title": "Transfert des licences",
      "body": ""
    },
    "feature6": {
      "title": "Fluidité de la création d'équipes",
      "body": ""
    }
  }
}