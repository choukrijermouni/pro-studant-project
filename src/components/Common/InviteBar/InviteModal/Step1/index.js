import {
  TypographyElement,
  Icon,
  Icons,
  Avatar,
  AvatarSize,
  Button,
  ButtonVariation,
  ButtonSize,
  SearchableSelectList
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import { invitation } from 'assets'
import { avatarName } from 'helpers'
import { useDispatch, useSelector } from 'react-redux'
import { annualLicenseType, billingCycleId, contactB2B, monthlyLicenseType, trialLicenseType } from 'constants/'
import { fetchListTeamsAction } from 'pages/Teams/store'
import { useEffect, useState } from 'react'
import { fetchLicenseInfoAction } from 'pages/Home/store'
import { inviteLearnersAction } from 'pages/Learners/store'

const useStyles = createUseStyles(style)
const noLoader = true

export default ({ nextStep, editLearners, closeModal, learners, subscriptionType, setSubscriptionType, setSelectedTeam, selectedTeam }) => {
  const dispatch = useDispatch()
  const {
    modal,
    avatars,
    invitesBox,
    invites,
    icon,
    header,
    userImage,
    editIcon,
    licenseButtons,
    line,
    circle,
    input,
    footer,
    noLicenseMessageContainer,
    image
  } = useStyles()
  const teams = useSelector(state => state.teams).list
  useEffect(() => {
    dispatch(fetchListTeamsAction())
    dispatch(fetchLicenseInfoAction(noLoader))
  }, [])
  const { licenseInfo } = useSelector(state => state.organization)
  const LicenseDetails = (licenseType) => licenseInfo?.OrganizationLicenseDetails?.find(license => license?.OrganizationLicenseType === licenseType)
  const reformedData = teams.map(element => {
    element.Value = element.Name
    return element
  })
  reformedData.push({
    Id: 0,
    Value: Text.withoutTeam
  })
  const [selectData, setSelectData] = useState({ selectedItemId: 1, selectedItemValue: '', reformedData: reformedData })
  const licenseAvailable = (
    ((LicenseDetails(monthlyLicenseType)?.LicenseTypeRemaining && LicenseDetails(monthlyLicenseType)?.LicenseTypeRemaining !== 0) ||
      (LicenseDetails(annualLicenseType)?.LicenseTypeRemaining && LicenseDetails(annualLicenseType)?.LicenseTypeRemaining !== 0) ||
      (LicenseDetails(trialLicenseType)?.LicenseTypeRemaining && LicenseDetails(trialLicenseType)?.LicenseTypeRemaining !== 0))
  )
  const licenseExists = (
    ((LicenseDetails(monthlyLicenseType)?.LicenseQuantity && LicenseDetails(monthlyLicenseType)?.LicenseQuantity !== 0) ||
      (LicenseDetails(annualLicenseType)?.LicenseQuantity && LicenseDetails(annualLicenseType)?.LicenseQuantity !== 0) ||
      (LicenseDetails(trialLicenseType)?.LicenseQuantity && LicenseDetails(trialLicenseType)?.LicenseQuantity !== 0))
  )
  return (
    <div>
      <div className={modal}>
        <div className={header}>
          <img className={image} src={invitation} alt='invitation' />
          <TypographyElement
            component='h2'
            variant='heading2'
            align='left'
            spacing='20px 0 8px 0'
          >
            {Text.inviteTitle}
          </TypographyElement>
        </div>
        <div>
          <TypographyElement
            component='h3'
            variant='heading3'
            align='left'
            spacing='8px 0 16px 0'
          >
            {Text.subHeader.first}{learners.length} {Text.subHeader.second}
          </TypographyElement>
          <div className={invitesBox}>
            <div className={invites}>
              <Icon iconName={Icons.admin} style={icon} />
              <TypographyElement
                component='h3'
                variant='heading3'
                align='left'
                spacing='0 200px 0px 27px'
              >
                {Text.learner}
              </TypographyElement>
            </div>
            <div className={avatars}>
              {learners.slice(0, 4).map((element, id) =>
                <div key={id} className={userImage}>
                  <Avatar
                    img={null}
                    name={avatarName(element)}
                    size={AvatarSize.size_1}
                    cursor='default'
                  />
                </div>)}
              <div className={circle}>{learners.length}</div>
              <Icon
                iconName={Icons.edit}
                style={editIcon}
                handleIconClick={editLearners}
              />
            </div>
          </div>
          <TypographyElement
            component='h3'
            variant='heading3'
            align='left'
            spacing='24px 0 16px 0'
          >
            {Text.licenseAffect}
          </TypographyElement>
          <div className={licenseButtons}>
            <Button
              dataTest='without-license-button'
              backgroundColorHover='none'
              variation={subscriptionType === billingCycleId.without ? ButtonVariation.primary : ButtonVariation.secondary}
              size={ButtonSize.big}
              width={128}
              height={70}
              label={Text.withoutButton}
              handleClick={() => setSubscriptionType(billingCycleId.without)}
            />
            <Button
              dataTest='trial-license-button'
              backgroundColorHover='none'
              variation={subscriptionType === billingCycleId.trial ? ButtonVariation.primary : ButtonVariation.secondary}
              size={ButtonSize.big}
              width={128}
              height={70}
              label={Text.trialButton}
              handleClick={() => setSubscriptionType(billingCycleId.trial)}
              disabled={!LicenseDetails(trialLicenseType)?.LicenseTypeRemaining || LicenseDetails(trialLicenseType)?.LicenseTypeRemaining < learners.length}
            />
            <Button
              dataTest='annual-license-button'
              backgroundColorHover='none'
              variation={subscriptionType === billingCycleId.annual ? ButtonVariation.primary : ButtonVariation.secondary}
              size={ButtonSize.big}
              width={128}
              height={70}
              label={Text.yearlyButton}
              handleClick={() => setSubscriptionType(billingCycleId.annual)}
              disabled={!LicenseDetails(annualLicenseType)?.LicenseTypeRemaining || LicenseDetails(annualLicenseType)?.LicenseTypeRemaining < learners.length}
            />
            <Button
              dataTest='monthly-license-button'
              backgroundColorHover='none'
              variation={subscriptionType === billingCycleId.monthly ? ButtonVariation.primary : ButtonVariation.secondary}
              size={ButtonSize.big}
              width={128}
              height={70}
              label={Text.monthlyButton}
              handleClick={() => setSubscriptionType(billingCycleId.monthly)}
              disabled={!LicenseDetails(monthlyLicenseType)?.LicenseTypeRemaining || LicenseDetails(monthlyLicenseType)?.LicenseTypeRemaining < learners.length}
            />
          </div>
          <div className={footer}>
            {
              licenseAvailable
                ? (
                  <div>
                    <TypographyElement
                      component='h4'
                      variant='caption1'
                      align='left'
                      spacing='20px 0 0px 0'
                    >
                      {Text.monthlyLicenseLeft}{LicenseDetails(monthlyLicenseType)?.LicenseTypeRemaining || 0}
                    </TypographyElement>
                    <TypographyElement
                      component='h4'
                      variant='caption1'
                      align='left'
                      spacing='0px 0 0px 0'
                    >
                      {Text.yearlyLicenseLeft}{LicenseDetails(annualLicenseType)?.LicenseTypeRemaining || 0}
                    </TypographyElement>
                    <TypographyElement
                      component='h4'
                      variant='caption1'
                      align='left'
                      spacing='0px 0 15px 0'
                    >
                      {Text.trialLicenseLeft}{LicenseDetails(trialLicenseType)?.LicenseTypeRemaining || 0}
                    </TypographyElement>
                  </div>
                  )
                : (
                  <div className={noLicenseMessageContainer}>
                    <TypographyElement
                      component='h4'
                      variant='caption1'
                      align='left'
                      spacing='20px 0 15px 0'
                    >
                      {`${licenseExists ? Text.noLicenseLeftMessage : Text.noLicenseMessage} ${contactB2B.ContactName} ${Text.on} ${contactB2B.ContactEmail}  ${Text.or} ${contactB2B.ContactPhone} ${Text.forGettingNewLicenses}`}
                    </TypographyElement>
                  </div>
                  )
            }
            {
              teams.length
                ? (
                  <span className={input}>
                    <SearchableSelectList
                      dataTest='teams-select-list'
                      iconDataTest='teams-select-list-icon'
                      items={reformedData}
                      placeholder={Text.allTeams}
                      selectedItem={selectData.selectedItemValue}
                      name='teams'
                      gutterBottom={16}
                      onSelectedItem={(item) => {
                        setSelectData({
                          ...selectData,
                          selectedItemValue: selectData.reformedData.find((element) => element.Id === item.Id).Value,
                          selectedItemId: item.Id
                        })
                        setSelectedTeam(item.Id)
                      }}
                      handleChange={(e) => setSelectData({
                        ...selectData,
                        selectedItemValue: e.target.value,
                        toggleSuggestionList: (e.target.value !== '')
                      })}
                    />
                  </span>
                  )
                : null
            }
          </div>
          <hr className={line} />
          <div className={licenseButtons}>
            <Button
              variation={ButtonVariation.secondary}
              size={ButtonSize.big}
              width={153}
              height={54}
              label={Text.abortButton}
              handleClick={closeModal}
            />
            <Button
              dataTest='confirm-invite-button'
              variation={ButtonVariation.primary}
              size={ButtonSize.big}
              width={153}
              height={54}
              label={Text.confirmButton}
              handleClick={() => dispatch(inviteLearnersAction(learners, subscriptionType, selectedTeam, nextStep, closeModal))}
            />
          </div>
        </div>
      </div>
    </div>
  )
}
