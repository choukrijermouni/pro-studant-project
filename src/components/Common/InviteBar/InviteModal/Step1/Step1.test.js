import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import Step1 from '.'
configure({ adapter: new Adapter() })

describe('>>> Step1 --- Shallow Render Step1 component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<Step1 />)
    tree = renderer.create(
      <Step1 />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
