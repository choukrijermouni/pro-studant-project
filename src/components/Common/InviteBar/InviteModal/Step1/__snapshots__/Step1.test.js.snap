// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Step1 --- Shallow Render Step1 component +++ render correctly and match snapshot 1`] = `
<div>
  <div
    className="modal-0-2-1"
  >
    <div
      className="header-0-2-2"
    >
      <img
        alt="invitation"
        src={Object {}}
      />
      <p
        className="text-0-2-12 text-d0-0-2-13"
      >
        Inviter des apprenants
      </p>
    </div>
    <div>
      <p
        className="text-0-2-14 text-d0-0-2-15"
      >
        Vous avez ajouter 
        6
         
        nouveaux apprenants a votre organisme
      </p>
      <div
        className="invitesBox-0-2-5"
      >
        <div
          className="invites-0-2-6"
        >
          <span
            className="icon-admins-alph-iconfont icon-0-2-7"
          />
          <p
            className="text-0-2-16 text-d0-0-2-17"
          >
            Apprenants
          </p>
        </div>
        <div
          className="avatars-0-2-3"
        >
          <div
            className="userImage-0-2-4"
          >
            <span
              className="avatarGroup-0-2-18 avatarGroup-d0-0-2-19"
            >
              YS
            </span>
          </div>
          <div
            className="userImage-0-2-4"
          >
            <span
              className="avatarGroup-0-2-20 avatarGroup-d0-0-2-21"
            >
              MD
            </span>
          </div>
          <div
            className="userImage-0-2-4"
          >
            <span
              className="avatarGroup-0-2-22 avatarGroup-d0-0-2-23"
            >
              AK
            </span>
          </div>
          <div
            className="userImage-0-2-4"
          >
            <span
              className="avatarGroup-0-2-24 avatarGroup-d0-0-2-25"
            >
              AH
            </span>
          </div>
          <div
            className="circle-0-2-11"
          >
            6
          </div>
          <span
            className="icon-edit-alph-iconfont editIcon-0-2-8"
          />
        </div>
      </div>
      <p
        className="text-0-2-26 text-d0-0-2-27"
      >
        Affecter une licence a vos apprenants
      </p>
      <div
        className="licenseButtons-0-2-9"
      >
        <button
          className="button-0-2-28 button-d0-0-2-30"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-32 text-d0-0-2-33 text-0-2-29 text-d1-0-2-31"
          >
            Annuel
          </p>
        </button>
        <button
          className="button-0-2-34 button-d0-0-2-36"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-38 text-d0-0-2-39 text-0-2-35 text-d1-0-2-37"
          >
            Mensuel
          </p>
        </button>
      </div>
      <p
        className="text-0-2-40 text-d0-0-2-41"
      >
        Licences Mensuelles restantes : 
      </p>
      <p
        className="text-0-2-42 text-d0-0-2-43"
      >
        Licences Annuel restantes : 
      </p>
      <hr
        className="line-0-2-10"
      />
      <div
        className="licenseButtons-0-2-9"
      >
        <button
          className="button-0-2-44 button-d0-0-2-46"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-48 text-d0-0-2-49 text-0-2-45 text-d1-0-2-47"
          >
            Annuler
          </p>
        </button>
        <button
          className="button-0-2-50 button-d0-0-2-52"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-54 text-d0-0-2-55 text-0-2-51 text-d1-0-2-53"
          >
            Confirmer
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
